src/openforms/prefill/contrib/haalcentraal/plugin.py

```diff
@@ -152,17 +153,40 @@ def get_co_sign_values(
         )
 
     def check_config(self):
+        """
+        The purpose of this fuction is to simply check the connection to the
+        service, so we are using dummy data and an endpoint which does not exist.
+        We want to avoid calls to the national registration by using bsn numbers.
+        """
+        config = HaalCentraalConfig.get_solo()
+        assert isinstance(config, HaalCentraalConfig)
+        if not config.service:
+            raise InvalidPluginConfiguration(_("Service not selected"))
+
+        client_errors = {
+            HaalCentraalVersion.haalcentraal13: 404,
+            HaalCentraalVersion.haalcentraal20: 400,
+        }
+
+        client = config.service.build_client()
         try:
-            config = HaalCentraalConfig.get_solo()
-            assert isinstance(config, HaalCentraalConfig)
-            if not config.service:
-                raise InvalidPluginConfiguration(_("Service not selected"))
-
-            client = config.service.build_client()
-            # FIXME: haal centraal v2 client does not support GET methods
-            client.retrieve("test", "test")
+            if config.version == HaalCentraalVersion.haalcentraal13:
+                client.retrieve("test", "test")
+            else:
+                client.operation(
+                    "test",
+                    data={
+                        "type": "RaadpleegMetBurgerservicenummer",
+                        "burgerservicenummer": ["test"],
+                        "fields": [AttributesV2.burgerservicenummer],
+                    },
+                    url="test",
+                    request_kwargs={
+                        "headers": {"Content-Type": "application/json; charset=utf-8"}
+                    },
+                )
```

Comment:
While working on a suggestion with `match / case`, it dawned on me that we can make this a lot simpler and rely on type checking to protect against future versions:

```suggestion
            client.make_config_test_request()
```

and then implement that method for both versions of the client + add it to `HaalCentraalClient` protocol to enforce future versions to imlement this from the get-go.

I would then implement that for the V1.3 client as:

```python
def make_config_test_request(self):
    try:
        self.retrieve("test", "test")
    except ClientError as e:
        if e.args[0].get("status") == 404:
            return
        raise
```
-> so for the expected 404 the test passes, otherwise it bubbles up to the plugin config check.